src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,377 @@
+            new GetDatesTestCase(true, List.of(QUERY_DATE /* Thursday */, "2020-08-14", "2020-08-15", "2020-08-16", "2020-08-17", "2020-08-18", "2020-08-19"))
+        );
+    }
+
+    /**
+     * Check that the ignoreRealtime query parameter is set to true
+     * regardless of whether it was originally missing or false.
+     */
+    @Test
+    public void testAddIgnoreRealtimeParam() throws URISyntaxException {
+        String queryWithRealtimeParam = BASE_QUERY + "&" + IGNORE_REALTIME_UPDATES_PARAM + "=false";
+        List<String> queries = List.of(BASE_QUERY, queryWithRealtimeParam);
+
+        for (String query : queries) {
+            MonitoredTrip trip = new MonitoredTrip();
+            trip.queryParams = query;
+            Map<String, String> params = ItineraryUtils.excludeRealtime(trip.parseQueryParams());
+            Assertions.assertEquals("true", params.get(IGNORE_REALTIME_UPDATES_PARAM));
+        }
+    }
+
+    /**
+     * Check that a given trip is modified with an OTP itinerary that exists,
+     * right now based on the index of the selected itinerary in the UI.
+     */
+    @Test
+    public void testUpdateTripWithVerifiedItinerary() throws URISyntaxException {
+        MonitoredTrip trip = new MonitoredTrip();
+        trip.id = "testUpdateTripWithVerifiedItinerary";
+        trip.queryParams = BASE_QUERY + "&ui_activeItinerary=1";
+        trip.itinerary = null;
+
+        List<Itinerary> itineraries = otpDispatcherPlanResponse.getResponse().plan.itineraries;
+
+        ItineraryUtils.updateTripWithVerifiedItinerary(trip, itineraries);
+
+        Assertions.assertEquals(itineraries.get(1), trip.itinerary);
+    }
+
+    @ParameterizedTest
+    @MethodSource("createSameDayTestCases")
+    void testIsSameDay(SameDayTestCase testCase) throws URISyntaxException {
+        MonitoredTrip trip = makeTestTrip(testCase.isArrival);
+        // The time zone for trip (and testCase.tripTime) is US Eastern per trip location.
+        ZoneId zoneId = trip.timezoneForTargetLocation();
+
+        Itinerary itinerary = simpleItinerary(testCase.tripTime, testCase.isArrival);
+        Assertions.assertEquals(
+            testCase.shouldBeSameDay,
+            ItineraryUtils.isSameDay(itinerary, QUERY_DATE, testCase.timeOfDay, zoneId, testCase.isArrival),
+            testCase.getMessage(zoneId)
+        );
+    }
+
+    private static List<SameDayTestCase> createSameDayTestCases() {
+        return List.of(
+            // Same-day departures
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__03_00_00_EDT, false, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__23_59_59_EDT, false, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__02_59_59_EDT, false, true),
+            new SameDayTestCase("1:23", _2020_08_12__03_00_00_EDT, false, true),
+            new SameDayTestCase("1:23", _2020_08_12__23_59_59_EDT, false, true),
```

Comment:
Since OTP shouldn't ever return a "depart at" itinerary that begins before the date + time request parameters, I don't believe such a case would occur. I guess this is what I'd expect for being the same day though, so I guess it's fine to keep it.